pkg/agentprofile/agent_profile.go

```diff
@@ -111,12 +111,18 @@ func ComponentOverrideFromProfile(profile *datadoghqv1alpha1.DatadogAgentProfile
 		Name:      profile.Name,
 	})
 
-	return v2alpha1.DatadogAgentComponentOverride{
+	profileComponentOverride := v2alpha1.DatadogAgentComponentOverride{
 		Name:       &overrideDSName,
 		Affinity:   affinityOverride(profile),
 		Containers: containersOverride(profile),
 		Labels:     labelsOverride(profile),
 	}
+
+	if priorityClassName := priorityClassNameOverride(profile); priorityClassName != nil {
```

Comment:
nit - this function is called `ComponentOverrideFromProfile` while the `priorityClassNameOverride` specifically checks `NodeAgentComponentName` which is confusing. Since we know that for foreseeable future DAP will work only for Agent component maybe we could drop/replace `Component` from the names.